### PR TITLE
Add RVV version of vpoc f32-vbinary

### PR DESCRIPTION
**Pull request by bhbruce**
This PR is to add RVV version of vopc f32-binary.

To fully test RVV version of f32-binary, this PR also introduces `vector_tile` to `tools/generate-vbinary-test.py`.

### CODE REVIEW

**Comment by bhbruce** [comment]
@fbarchard @alankelly If you have free time, please help to review this.
cc @phoebesv @rednoah91

File: src/f32-vbinary/vopc-rvv.c.in [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  $if ACTIVATION == "MINMAX":
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
+    vfloat32m${LMUL}_t voutput_min = __riscv_vfmv_v_f_f32m${LMUL}(params->scalar.min, vl);
+    vfloat32m${LMUL}_t voutput_max = __riscv_vfmv_v_f_f32m${LMUL}(params->scalar.max, vl);
+  $elif ACTIVATION == "RELU":
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
+    vfloat32m${LMUL}_t voutput_min = __riscv_vfmv_v_f_f32m${LMUL}(0.0f, vl);
+
+  do {
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
```
Comment (fbarchard) [inline src/f32-vbinary/vopc-rvv.c.in:54]: consider moving this to bottom of loop and ensure the first one is done outside of loop for MINMAX/RELU and else (LINEAR)

File: src/f32-vbinary/vopc-rvv.c.in [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    n -= vl;
+    vfloat32m${LMUL}_t va = __riscv_vle32_v_f32m${LMUL}(input_a, vl);
+    input_a += vl;
+    vfloat32m${LMUL}_t vacc = ${OP_FUNC}m${LMUL}(va, b, vl);
+    $if OP == "SQRDIFF":
+      vacc = __riscv_vfmul_vv_f32m${LMUL}(vacc, vacc, vl);
+    $if ACTIVATION == "MINMAX":
+      vacc = __riscv_vfmax_vv_f32m${LMUL}(vacc, voutput_min, vl);
```
Comment (fbarchard) [inline src/f32-vbinary/vopc-rvv.c.in:62]: use  __riscv_vfmax_vf_f32m etc for mul, max, min against a scalar.  
Which also eliminates the need to set vectors/vlen outside of loop

**Review by fbarchard — CHANGES_REQUESTED** [review]
if you can, provide a benchmark indicating vectorization helped and which LMUL was fastest

File: src/f32-vbinary/vopc-rvv.c.in (hunk 2 above)
Comment (bhbruce) [reply src/f32-vbinary/vopc-rvv.c.in]: I've changed to use `__riscv_vfmax_vf_f32m ` in latest push.

File: src/f32-vbinary/vopc-rvv.c.in [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  do {
+    size_t vl = __riscv_vsetvl_e32m${LMUL}(n);
+    n -= vl;
+    vfloat32m${LMUL}_t va = __riscv_vle32_v_f32m${LMUL}(input_a, vl);
+    input_a += vl;
+    vfloat32m${LMUL}_t vacc = ${OP_FUNC}m${LMUL}(va, b, vl);
+    $if OP == "SQRDIFF":
+      vacc = __riscv_vfmul_vv_f32m${LMUL}(vacc, vacc, vl);
+    $if ACTIVATION == "MINMAX":
+      vacc = __riscv_vfmax_vf_f32m${LMUL}(vacc, params->scalar.min, vl);
```
Comment (fbarchard) [inline src/f32-vbinary/vopc-rvv.c.in:55]: normally we'd put params into const floats outside the loop, not depending on the compiler being smart enough to houst the loads.
its fine as is... maybe just check the code gen is what you're hoping for

**Review by fbarchard — APPROVED** [review]

File: src/f32-vbinary/vopc-rvv.c.in (hunk 3 above)
Comment (bhbruce) [reply src/f32-vbinary/vopc-rvv.c.in:55]: Thank you for providing this trick.
I'll change to use `const floats outside the loop`.

Comment (bhbruce) [reply src/f32-vbinary/vopc-rvv.c.in]: Done.

**Pull request merged**